lib/rules/valid-typeof.js

```diff
@@ -74,6 +96,20 @@ module.exports = {
                             }
                         } else if (requireStringLiterals && !isTypeofExpression(sibling)) {
                             context.report({ node: sibling, messageId: "notString" });
+                        } else if (sibling.type === "Identifier" && sibling.name === "undefined" && isReferenceToGlobalVariable(sibling)) {
+                            context.report({
+                                node: sibling,
+                                messageId: "invalidValue",
+                                data: { type: "undefined" },
+                                suggest: [
+                                    {
+                                        messageId: "suggestString",
+                                        fix(fixer) {
+                                            return fixer.replaceText(sibling, '"undefined"');
+                                        }
+                                    }
+                                ]
+                            });
```

Comment:
```suggestion
                                messageId: "invalidValue",
                                suggest: [
                                    {
                                        messageId: "suggestString",
                                        data: { type: "undefined" },
                                        fix(fixer) {
                                            return fixer.replaceText(sibling, '"undefined"');
                                        }
                                    }
                                ]
                            });
```

Suggestions have their own `data` property (https://eslint.org/docs/developer-guide/working-with-rules#placeholders-in-suggestion-messages).

It currently shows `` Use `"{{type}}"` instead of `{{type}}`. `` because the top-level `data` isn't used for suggestions so the placeholders aren't replaced:

![image](https://user-images.githubusercontent.com/44349756/156163401-69c1b156-dab0-4d84-94ee-0762dfbbf9b8.png)